src/worker-thread/initialize.ts

```diff
@@ -23,15 +23,17 @@ import { createStorage } from './Storage';
 import { StorageLocation } from '../transfer/TransferrableStorage';
 import { appendGlobalEventProperties } from './dom/HTMLElement';
 
+export type WorkerStorageInit = { supported: true; storage: { [key: string]: string } } | { supported: false; errorMsg: string };
```

Comment:
This does mean the return type is different for supported or not... could we simplify this?

```suggestion
export type WorkerStorageInit = {
  storage: null | { [key: string]: string }
  error: null | string;
}
```

If storage is not `null` use the value, otherwise output the `error`.

Reply:
From a domain modeling perspective, that would imply that it is possible for `{storage: null, error: null}` object to exist. When possible, I'd prefer to make incorrect states impossible to even represent. 

I do see your point though that `success` can be implicit and derived from the existence of storage or error. I'm not sure how to accomplish that without providing a shared key to discriminate on.

Reply:
I'm mostly concerned about the shape of the return object being different given inputs.

Another idea:

```suggestion
export type WorkerStorageInit = { 
  storage: { [key: string]: string };
  error: null;
} | {
  storage: null;
  error: string;
};
```

Reply:
Ah! figured it out with the `in` operator for type narrowing. Pushing an update now

Reply:
>I'm mostly concerned about the shape of the return object being different given inputs.

For v8 reasons? I like your latest shape, updating now

Reply:
> For v8 reasons? I like your latest shape, updating now

Yep, it's super minor here but in general keeping the same shape is preferred for a method return value (afaik).